components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.endpoint/src/main/java/org/wso2/carbon/identity/configuration/mgt/endpoint/impl/SearchApiServiceImpl.java

```diff
@@ -42,14 +44,20 @@
 public class SearchApiServiceImpl extends SearchApiService {
 
     private static final Log LOG = LogFactory.getLog(SearchApiServiceImpl.class);
+    private boolean allowCrossTenantSearch = Boolean.parseBoolean(IdentityUtil.getProperty("ConfigurationStore" +
+            ".AllowCrossTenantSearch"));
 
     @Override
     public Response searchGet(SearchContext searchContext) {
 
         try {
-            Resources resources = getConfigurationManager().getTenantResources(
-                    getSearchCondition(searchContext)
-            );
+            Resources resources;
+            if (allowCrossTenantSearch) {
+                resources = getConfigurationManager().getTenantResources(getSearchCondition(searchContext));
+            } else {
+                resources = getConfigurationManager().getTenantResources(IdentityTenantUtil
+                                .getTenantDomainFromContext(), getSearchCondition(searchContext));
```

Comment:
If tenant qualified URL is not enabled will this have an impact?
IIRC, the thread-local gets populated on tenant qualified mode only, whereas this API is implemented to be accessible from a tenant in the path in the legacy way

Reply:
This worked as expected even when tenant qualified URL mode is disabled.